src/iterator.vala

```diff
@@ -58,37 +48,26 @@ public class Vast.Iterator : Object
     public Value
     get_value ()
     {
-        return array.get_value (_cursor);
+        return memory_to_value(get(), array.scalar_type, array.scalar_size);
     }
 
     public void
-    set (void* val)
+    set (void* memory)
     {
-        Memory.copy ((uint8*) array.data.get_data () + _offset, val, array.scalar_size);
+        Memory.copy (get(), memory, array.scalar_size);
```

Comment:
Yes. We can avoid function calls here all together. get_data() looks unncessary too.